pywinauto/linux/atspi_objects.py

```diff
@@ -433,6 +530,21 @@ class _AtspiAccessible(Structure):
 ]
 
 
+def g_error_handler(func):
+    def wrapper(*args, **kwargs):
+        error = _GError()
+        ep = POINTER(POINTER(_GError))(error)
+        kwargs["g_error_pointer"] = ep
+        res = func(*args, **kwargs)
+        if ep and ep.contents:
+            print("Warning g_error_handled during {} call. "
+                  "Error code: {}. "
+                  "Error message: {}".format(func.__name__, ep.contents.contents.code, ep.contents.contents.message))
+        return res
+
+    return wrapper
+
```

Comment:
I actually, came up with a similar decorator in my AtspiDocument branch (still WIP). We need ```@wraps(func)``` decorator from ```functools```. This is to prevent loosing the information about the wrapped function. I also think it should throw an exception. Something similar to "COMError" we have in UIA backend. 

```python
def _handle_gerror(func, exception=ValueError):
    """Helper decorator to handle GError"""

    @wraps(func)
    def wrapper(self, *args, **kwargs):
        """Handle the call result according to returned GError"""
        err_p = pointer(_GError())
        err_pp = pointer(err_p)
        res = func(self, err_pp, *args, **kwargs)
        if err_p[0].code == 0:
            return res
        else:
            raise exception("GError with code: {0}, message: '{1}'".format(
                            err_p[0].code, err_p[0].message.decode(encoding='UTF-8')))
    return wrapper
```

Reply:
BTW, this decorator might be leaking memory as GError is not freed after handling, but we can deal with it later. Still need to figure out a way to get actual error.